sys/pm_callouts.c

```diff
@@ -408,7 +408,161 @@ void redir(portmaster_packet_info* packetInfo, portmaster_packet_info* redirInfo
     return;
 }
 
-void send_tcp_rst(portmaster_packet_info* packetInfo, void* originalPacket, ULONG originalPacketLength) {
+void send_icmp_blocked_packet(portmaster_packet_info* packetInfo, void* originalPacket, ULONG originalPacketLength, BOOL useLocalHost) {
+    // Only UDP is supported
+    if(packetInfo->protocol != 17) { // 17 -> UDP
+        return; // Not UDP
+    }
+
+    if(packetInfo->ipV6) {
+        // Initialize header for the original UDP packet
+        ULONG originalIPHeaderLength = calc_ipv6_header_size(originalPacket, originalPacketLength, NULL);
+        PIPV6_HEADER originalIPHeader = (PIPV6_HEADER) originalPacket;
+        UINT16 bytesToCopyFromOriginalPacket = (UINT16)originalPacketLength;
```

Comment:
Would it not be enough to just copy the header of the original packet?

Reply:
Yes the header should contain all the need information but the RFC says "As much of invoking packet as possible without the ICMPv6 packet exceeding the minimum IPv6 MTU".
https://www.rfc-editor.org/rfc/rfc4443#section-3.1

In my opinion is kind of risky to send just the header, it may not work with some devices or stop working in the feature.
I don't think the performance benefits will be that big as we don't send icmp packets that often.

Reply:
OK. I understand.